lib/active_admin/namespace.rb

```diff
@@ -226,8 +242,29 @@ def unload_resources!
 
     # Creates a ruby module to namespace all the classes in if required
     def register_module
-      unless Object.const_defined? module_name
-        Object.const_set module_name, Module.new
+      # dynamically create nested modules
+      module_names = module_name.split("::").inject([]) { |n, c| n << (n.present? ? [n.last] + [c] : [c]).flatten }
+      module_names.each_with_index do |module_name_array, i|
+        *prefix, parent, child = module_name_array
```

Comment:
My guess is this can be simplified by extracting a `parent_module` method that is called recursively.

Reply:
Sure, I will think a bit on how to make it more general